docs/src/content/docs/get-started/url-shortner.mdx

```diff
@@ -0,0 +1,306 @@
+---
+title: "Building a URL Shortner"
+description: "Create a simple URL Shortner using DiceDB Go SDK."
+sidebar:
+  order: 5
+---
+
+This tutorial guides you through creating a URL shortener using DiceDB, a key-value store, with Go. We’ll set up endpoints to generate short URLs and redirect them to the original URLs.
+
+Prerequisites
+
+1. Go installed (at least version 1.18)
+2. DiceDB server running locally
+3. Basic familiarity with Go and DiceDB
+
+## Environment setup
+
+Refer to [DiceDB Installation Guide](https://dicedb.io/get-started/installation/) to get your DiceDB server up and running with a simple Docker command.
+
+## Starting the application server
+
+1. Clone the repository
+   ```bash
+   git clone https://github.com/Prachi-Jamdade/url-shortner-go-dicedb
+   cd url-shortner-go-dicedb
+   ```
+2. Start the application
+   ```bash
+   go run main.go
+   ```
+   This will start the application server on port 8080 by default, you should see output similar to
+   ```bash
+   [GIN-debug] Listening and serving HTTP on :8080
+   ```
+
+## Interacting with the application
+
+1. *Start DiceDB*: Ensure DiceDB is running.
+2. *Test the API*:
+    - Shorten URL:
+        Send a POST request to `/shorten` with JSON body on Postman:
+        ```
+        {
+            "long_url": "https://example.com"
+        }
+        ```
+
+        OR
+
+        ```curl
+        curl -X POST -H "Content-Type: application/json" 
+            -d '{"long_url": "https://example.com"}' http://localhost:8080/shorten
+        ```
+
+    - Redirect to Original URL:
+        Send a GET request to `/:id` with the short URL ID on Postman
+
+        OR
+
+        ```curl
+        curl -L http://localhost:8080/{short_id}
+        ```
+
+## DiceDB Commands Used
+
+Here are the main DiceDB commands we’ll use to store and retrieve URLs.
+
+1. `Set` Command: Stores a key-value pair in DiceDB.
+Syntax - `Set(key, value, expiration)`
+`key`: Unique identifier (e.g., short URL code)
+`value`: The data to store (serialized JSON)
+`expiration`: Optional; 0 means no expiration
+
+2. `Get` Command: Retrieves the value associated with a key.
+Syntax - `Get(key)`
+`key`: The identifier for the data to retrieve.
+
+## Project Structure
+
+The project is organized into several packages:
+
+- `models`: Defines the URL data structure.
+- `repository`: Handles database interactions with DiceDB.
+- `services`: Implements business logic for URL shortening.
+- `handlers`: Contains HTTP route handlers for the API.
+- `utils`: Provides helper functions for API responses.
+
+## Code overview
+
+1. `models/url.go`: Defines the URL struct, which represents the data to be stored in DiceDB.
+
+    ```go
+    package models
+
+    type URL struct {
+        ID       string `json:"id"`
+        LongURL  string `json:"long_url"`
+        ShortURL string `json:"short_url"`
+    }
```

Comment:
we don't need to define a model here. The key can be the short URL, the value can be the long URL.
Again, this tutorial should have the most basic setup (no need to think too deeply about extensibility)